src/components/select/select.ts

```diff
@@ -0,0 +1,377 @@
+  @property({ reflect: true, type: Boolean })
+  public outlined = false;
+
+  /** The autofocus attribute of the control. */
+  @property({ type: Boolean })
+  public override autofocus!: boolean;
+
+  /** The label attribute of the control. */
+  @property({ type: String })
+  public label!: string;
+
+  /** The placeholder attribute of the control. */
+  @property({ type: String })
+  public placeholder!: string;
+
+  /** Whether the dropdown's width should be the same as the target's one. */
+  @property({ type: Boolean, attribute: 'same-width' })
+  public override sameWidth = true;
+
+  /** The direction attribute of the control. */
+  @property({ reflect: true })
+  public override dir: 'ltr' | 'rtl' | 'auto' = 'auto';
+
+  constructor() {
+    super();
+    this.size = 'medium';
+  }
+
+  /** Sets focus on the component. */
+  public override focus(options?: FocusOptions) {
+    this.target.focus(options);
+  }
+
+  /** Removes focus from the component. */
+  public override blur() {
+    this.target.blur();
+  }
+
+  /** Checks the validity of the control. */
+  public reportValidity() {
+    this.invalid = this.required && !this.value;
+    if (this.invalid) this.focus();
+    return !this.invalid;
+  }
+
+  public override async firstUpdated() {
+    super.firstUpdated();
+    await this.updateComplete;
+
+    const selectedItem = this.allItems.find((i) => i.selected) ?? null;
+    if (selectedItem) this.value = selectedItem.value;
+    else this.updateSelected();
+  }
+
+  @watch('selectedItem')
+  protected updateValue() {
+    this.value = this.selectedItem?.value ?? this.value;
+  }
+
+  @watch('value')
+  protected updateSelected() {
+    if (this.allItems.length === 0) return;
+
+    if (this.selectedItem?.value !== this.value) {
+      const matches = this.allItems.filter((i) => i.value === this.value);
+      const index = this.allItems.indexOf(matches[matches.length - 1]);
+
+      if (index === -1) {
+        this.value = undefined;
+        this.clearSelection();
+        return;
+      }
+
+      this.select(index);
+    }
+  }
+
+  @watch('value')
+  protected validate() {
+    this.updateComplete.then(() => this.reportValidity());
+  }
+
+  protected selectNext() {
+    const activeItemIndex = [...this.allItems].indexOf(
+      this.selectedItem ?? this.activeItem
+    );
+
+    const next = this.getNearestSiblingFocusableItemIndex(
+      activeItemIndex ?? -1,
+      1
+    );
+    this.selectItem(this.allItems[next], true);
+  }
+
+  protected selectPrev() {
+    const activeItemIndex = [...this.allItems].indexOf(
+      this.selectedItem ?? this.activeItem
+    );
+    const prev = this.getNearestSiblingFocusableItemIndex(
+      activeItemIndex ?? -1,
+      -1
+    );
+    this.selectItem(this.allItems[prev], true);
+  }
+
+  protected selectFirstActiveItem() {
+    const items = this.allItems.filter((i) => !i.disabled);
+    const index = this.allItems.indexOf(items[0]);
+    const item = this.allItems[index];
```

Comment:
Seems like the index is not needed for anything else, so this can be reduced to (also consistent with dd logic):
```suggestion
    const item = this.allItems.filter((item) => !item.disabled).at(0);
```
Same goes for the last item below with `.at(-1)`. Also, on the off chance there are no items or something, do check with `if (item && item.value...)` to be safe